node/src/actors/chain_manager/mod.rs

```diff
@@ -966,42 +1056,102 @@ impl ChainManager {
+                act.chain_state.superblock_state.has_consensus()
+            } else {
+                // If the node is not synced yet, assume that the superblock is valid.
+                // This is because the node is consolidating blocks received during the synchronization
+                // process, which are assumed to be valid.
+                SuperBlockConsensus::SameAsLocal
             };
-            let ars_ordered_keys = &act.chain_state.last_ars_ordered_keys;
-
-            let superblock = act.chain_state.superblock_state.build_superblock(
-                &block_headers,
-                &ars_members,
-                ars_ordered_keys,
-                superblock_index,
-                last_hash,
-            );
-            actix::fut::ok(superblock)
+            match consensus {
+                SuperBlockConsensus::SameAsLocal => {
+                    // Consensus: persist chain state
+                    log::info!("Before update: superblock {:?}", act.get_superblock_beacon());
+                    act.chain_state.chain_info.as_mut().unwrap().highest_superblock_checkpoint =
+                        act.chain_state.superblock_state.get_beacon();
+                    if act.sm_state == StateMachine::Synced {
+                        act.persist_chain_state(ctx);
+                    }
+                    log::info!("Consensus! Superblock {:?}", act.get_superblock_beacon());
+                    log::info!("Current tip of the chain: {:?}", act.get_chain_beacon());
+                    log::info!(
+                        "The last block of the consolidated superblock is {}",
+                        last_hash
+                    );
+                    let chain_info = act.chain_state.chain_info.as_ref().unwrap();
+                    let ars_members = {
+                        // Before reaching the epoch activity_period + collateral_age the bootstrap committee signs the superblock
+                        // collateral_age is measured in blocks instead of epochs, but this only means that the period in which
+                        // the bootstrap committee signs is at least epoch activity_period + collateral_age
+                        if act.current_epoch.unwrap()
+                            > chain_info.consensus_constants.collateral_age
+                                + chain_info.consensus_constants.activity_period
+                        {
+                            act.chain_state.last_ars.clone()
+                        } else {
+                            chain_info
+                                .consensus_constants
+                                .bootstrapping_committee
+                                .iter()
+                                .map(|add| add.parse().expect("Malformed bootstrapping committee"))
+                                .collect()
+                        }
+                    };
+                    let ars_ordered_keys = &act.chain_state.last_ars_ordered_keys;
```

Comment:
are you taking into account the case that you evaluate in the upper lines (the "mengano" committee) for the bls keys?